src/settings/PathConfigStorage.cpp

```diff
@@ -7,6 +7,41 @@
 #include "../raft.h"
 #include "../sliceDataStorage.h" // SliceDataStorage
 
+namespace
+{
+
+// Check whether the given layer_nr is the first layer of support structure that touches the raft or build plate.
+bool isFirstSupportLayer(const cura::LayerIndex& layer_nr)
```

Comment:
Sorry, if something was confused you. Let me please explain.

- If raft was turned off, the first layer is always the layer 0, that's absolutely true. In this case, the layer 0 will be the first layer of support structures (that touches the buildplate).
- But if we have the raft turned on, then there may be some amount of filler layers - layers to fill the raft air gap and initial layer with helper parts (these are support layers that touch raft). 

The purpose of this method is to find the number of the very first support layer built on the top of the raft – first of the filler layer - if raft was turned on. Since raft layers have a negative value, for example, 
raft base = -7 (this is the raft layer that touches the buildplate, but we have to find the _first of support layer_ that touches the buildplate or raft)
raft middle = -6
raft top = -5
Then, before we goes print 0 layer, there are filler layers with support structures, and the first layer of support that touches raft will be -4.

Let me please know does it make sense.

Reply:
Oh, fair enough. I didn't realize that the support was not using the air_gap and it's printed right on top of the last layer of raft.